spec/Appendix B -- Grammar Summary.md

```diff
@@ -156,10 +156,22 @@ Selection :
 - FragmentSpread
 - InlineFragment
 
-Field : Alias? Name Arguments? Directives? SelectionSet?
+Field : Alias? Name Arguments? Nullability? Directives? SelectionSet?
 
 Alias : Name :
 
+Nullability :
+
+- ListNullability NullabilityDesignator?
+- NullabilityDesignator
+
+ListNullability : `[` Nullability? `]`
```

Comment:
Is there a good reason for the `?` here? 

This means that something like `field[[]]` is allowed, but doesn't add any value, right? Can it be argued that this may be confusing so it would make more sense to disallow it?

Reply:
Yep. `field[[]]` would technically be allowed but do nothing. The `?` is there because the nullability operator is optional. If it was not optional then `[[[!]]]?` would not be valid because the second dimension of the list doesn't have a nullability operator.

Reply:
To clarify, I am talking about the `?` in `ListNullability : [ Nullability? ]`

If it were `ListNullability : [ Nullability ]` instead, then `[[!]]?` would be valid, right? But `[[]]?` wouldn't, which I think is the preferable behavior.

Reply:
Doesn't it make sense to allow `[[]?]` though? It could be coded as `[?]` but I think it's better to be more explicit (here there's a list of lists, and the inner list is nullable) as that helps people reading the query. I even imagine I'd add lint rules to enforce this for clarity.

Reply:
>Doesn't it make sense to allow [[]?] though?
Thanks @benjie, you are right! 

So `` ListNullability : `[` Nullability? `]` `` does make sense.